src/qt/walletmodel.cpp

```diff
@@ -581,11 +581,12 @@ bool WalletModel::privateKeysDisabled() const
 bool WalletModel::canGetAddresses() const
 {
     // The wallet can provide a fresh address if:
-    // * hdEnabled(): an HD seed is present; or
+    // * hdEnabled(): a HD seed is present; or
     // * it is a legacy wallet, because:
-    //     * !hdEnabled(): an HD seed is not present; and
-    //     * !IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS): private keys have not been disabled (which results in hdEnabled() == true)
-    return m_wallet->hdEnabled() || (!m_wallet->hdEnabled() && !m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS));
+    //     * !hdEnabled(): a HD seed is not present; and
+    //     * !IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS): private keys have not been disabled
+    //     * !isBlank(): the wallet is not blank
+    return m_wallet->hdEnabled() || (!m_wallet->hdEnabled() && !m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS) && !m_wallet->isBlank());
```

Comment:
In commit "[wallet] Support creating a blank wallet" (8061ea4611696b22d5d5dcdd73c1e053ffecc09d)

I think this entire line is equivalent to `return !m_wallet->isBlank()`.